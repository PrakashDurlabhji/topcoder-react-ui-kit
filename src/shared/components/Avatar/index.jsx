import DefaultUserAvatar from 'assets/images/default-user-avatar.svg';
import React from 'react';
import { themr } from 'react-css-super-themr';
import { Avatar as ProtoAvatar } from 'topcoder-react-utils';

import theme from './style.scss';

function Avatar(props) {
  return <ProtoAvatar DefaultAvatar={DefaultUserAvatar} {...props} />;
}

export default themr('Avatar', theme)(Avatar);
