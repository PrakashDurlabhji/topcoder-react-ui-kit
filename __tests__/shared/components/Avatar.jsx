import Avatar from 'components/Avatar';
import React from 'react';
import { snapshot } from 'topcoder-react-utils/jest-utils';

test('Snapshots', () => {
  snapshot(<Avatar />);
});
