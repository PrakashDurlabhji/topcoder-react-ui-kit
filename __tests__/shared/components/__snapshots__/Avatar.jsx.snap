// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots 1`] = `
<svg
  className="style__avatar___3QhkH"
  height="141"
  viewBox="0 0 141 141"
  width="141"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <title>
    ico-user-default
  </title>
  <defs>
    <rect
      height="140"
      id="a"
      rx="306"
      width="140"
      x=".8"
      y=".5"
    />
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <mask
      fill="#fff"
      id="b"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
    <use
      fill="#F0F0F0"
      xlinkHref="#a"
    />
    <path
      d="M118 137.143c0 4.345-3.501 7.857-7.833 7.857H31.833C27.51 145 24 141.488 24 137.143c0-15.714 15.142-30.376 30.62-36.174-8.937-5.54-14.953-15.377-14.953-26.683v-7.857C39.667 49.072 53.697 35 71 35c17.304 0 31.333 14.072 31.333 31.429v7.857c0 11.306-6.016 21.143-14.946 26.683 15.471 5.798 30.613 20.46 30.613 36.174z"
      fill="#A3A3AE"
      mask="url(#b)"
      opacity=".2"
      stroke="#A3A3AE"
      strokeWidth="3"
    />
  </g>
</svg>
`;
